### ISSUE TITLE
Struggling to connect at all

### ISSUE THREAD

**Issue by chrisclaynz** [bromagosa/Snap4Arduino-plugin#5]
Connected inconsistently via Bluetooth. When I tried to see if usb connection was more consistent I was unable to even find the usb port when clicking "connect arduino". The bluetooth option remained despite the module being disconnected and unpaired.

I'm working it with a Mac.

**Comment by bromagosa** [comment]
Have you tried with a cable instead of bluetooth? This version is still being worked on, connecting via bluetooth is adding yet another possible source of issues and it's hard to narrow down the problem...

In any case, when everything fails, try:

- Go to ``chrome://extensions``
- Click on the ``background.html`` link under the Snap4Arduino connector extension
- Paste a screenshot of what you see, in case it can help me find out what the problem is
- Disconnect and reconnect the board
- Refresh that page (the one that popped up when clicking on background.html)
- Refresh the Snap4Arduino page too

Now you should be able to connect.

**Comment by chrisclaynz** [comment]
Hey Bernat.

I've just had another play (with bluetooth again since I didn't have a
cable around).

Anyway - here is the screenshot as requested. I hope it's useful in some
way.

Cheers,

Chris

**Comment by chrisclaynz** [comment]
I just connected and it seemed to work well.

However, when I used my custom block that was created with our firmata (
https://github.com/chrisclaynz/Robotics-unleashed) I cannot get our
Ultrasound blocks to read...

It gives the following error on the blocks:

Inside: TypeError
this.arduino.board.once is not a function

Any ideas..? Maybe I'm driving it too hard?

Cheers,

Chris

**Comment by bromagosa** [comment]
Yep, definitely driving it too hard hehe. The ``board`` object, and its ``sp`` object, are just proxies to the "real" ones, usable only from the browser extension side. Here's the list of currently proxied board methods at the client side, and I'm afraid I haven't added the ``once`` method yet:

![image](https://cloud.githubusercontent.com/assets/1016697/16190323/ecbc3df8-36de-11e6-9693-a2138227e1fd.png)

These methods are actually just sending a message to the browser extension side:

``
function (pin, callback) { postal.sendCommand('analogRead', [ board.id, pin ], callback) }
``

If JavaScript was just a little bit more flexible and implemented a ``DoesNotUnderstand`` pattern, I would be able to proxy everything in a transparent way, but unfortunately as it is I need to proxy each single method by hand :(

Give me a couple hours to finish fixing some issues in the regular Snap4Arduino version and I'll get back to this, it can't be too hard to implement it.

Thanks for your patience and testing!

(By the way, I'm afraid your screenshot was not attached, you may need to use the GitHub interface to post images)

**Comment by seteahconnoc** [comment]
Dear bromagosa,

thank you for the Snap4Arduino-plugin. I am currently using it for a project on a Raspberry Pi 3 Model B. Right now, I want to connect the ultrasound sensor HC-SR04 according to this library:

https://github.com/mdcanham/Snap4Arduino-Firmata/tree/master/Ultrasound_HC_SR04Firmata

Sadly, I am getting the error:

Inside: TypeError
this.arduino.board.once is not a function

Do you have any advice? Or do you happen to know any other way to use an ultrasound sensor with you plugin.

Thank you in advance.

**Comment by bromagosa** [comment]
Hi @seteahconnoc!

That's related to issue #7, but please note that this repository is about to be deprecated and closed, as all Snap4Arduino versions are now being built by a single script, and the code for all of them is at the main [Snap4Arduino repo](http://github.com/bromagosa/Snap4Arduino).

Anyway, this should hopefully have been solved in version 1.2. Please visit http://snap4arduino.org and try it out :)
